solutions/ng-rx/src/app/app.module.ts

```diff
@@ -0,0 +1,23 @@
+
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { HttpModule } from '@angular/http';
+import { provideStore, Store, StoreModule } from '@ngrx/store';
+import { AppComponent } from './app.component';
+import { EffectsModule } from '@ngrx/effects';
+
+import { TodoInputComponent } from './components/todo-input.component';
+import { TodoListComponent } from './components/todo-list.component';
+import { TodoReducer } from './store/Todo.reducer';
+import { TodoEffects } from './store/todo.effect';
+import { TodoActions } from './store/todo.action';
+import { TodoService } from './services/todo.service';
+
+@NgModule({
+  declarations: [AppComponent, TodoInputComponent, TodoListComponent],
+  imports: [BrowserModule, FormsModule, HttpModule, StoreModule.provideStore({ TodoReducer }), EffectsModule.run(TodoEffects)],
```

Comment:
provideStore takes an object that includes an identifier as key and reducer as value.  It's a little weird for the `Reducer` suffix to be used for the identifier of the store, since we would be selecting a collection of data, and not a reducer.  If we call the property `todoList` I think it will read better when we select it.  @ngrx provides an example like this:

```
import { NgModule } from '@angular/core'
import { StoreModule } from '@ngrx/store';
import { counterReducer } from './counter';

@NgModule({
  imports: [
    BrowserModule,
    StoreModule.provideStore({ counter: counterReducer })
  ]
})
export class AppModule {}
```